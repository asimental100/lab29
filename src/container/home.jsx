import React, { Component } from 'react';
import { fetches } from '../services/fetches';
import { Header } from '../components/header/header';
import { Search } from '../search/search';
import { Responses } from '../responses/responses';
import styles from './home.css';

export class Home extends Component {
  state = {
    url: "",
    method: "",
    response: {},
    json: {}
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleClick = async() => {
    const { url, method, json } = this.state;
    
    const response = await fetches(url, method, json);
    this.setState({ response });
  };

  render() {
    const { response } = this.state;
    return (
      <div className={styles.mainDiv}>
        <Header />
        <Search onChange={this.handleChange} onClick={this.handleClick} />
        <Responses response={{response}} />
      </div>      
    )
  }
}
