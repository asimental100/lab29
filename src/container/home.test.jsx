import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Home } from './home';

describe('Home component', () => {
  afterEach(() => cleanup());

  it('should render Home component', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
