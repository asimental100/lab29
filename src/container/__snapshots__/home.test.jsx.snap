// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App compontent should render App 1`] = `
<DocumentFragment>
  <div
    class="mainDiv"
  >
    <h1
      class="header"
    >
      Postman Clone
    </h1>
    <div
      class="mainDiv"
    >
      <div
        class="search"
      >
        <label
          for="url"
        >
          <h1>
            Search
          </h1>
        </label>
        <input
          name="url"
          placeholder="Search"
          type="text"
        />
      </div>
      <div>
        <label
          for="method"
        >
          <h1>
            Methods
          </h1>
        </label>
        <ul
          class="methodsList"
        >
          <li>
            GET
            <input
              name="method"
              type="radio"
              value="GET"
            />
          </li>
          <li>
            POST
            <input
              name="method"
              type="radio"
              value="POST"
            />
          </li>
          <li>
            PUT
            <input
              name="method"
              type="radio"
              value="PUT"
            />
          </li>
          <li>
            PATCH
            <input
              name="method"
              type="radio"
              value="PATCH"
            />
          </li>
          <li>
            DELETE
            <input
              name="method"
              type="radio"
              value="DELETE"
            />
          </li>
        </ul>
      </div>
      <div
        class="json"
      >
        <label
          for="json"
        >
          <h3>
            JSON (POST/PUT)
          </h3>
        </label>
        <input
          name="json"
          placeholder="JSON"
          type="text"
        />
      </div>
      <button>
        Send!
      </button>
    </div>
    <div
      class="display"
    >
      <h1>
        Response
      </h1>
      <div
        class="react-json-view"
        style="font-family: monospace; cursor: default; background-color: rgb(3, 26, 22); position: relative;"
      >
        <div
          class="pretty-json-container object-container"
        >
          <div
            class="object-content"
          >
            <div
              class="object-key-val"
            >
              <span>
                <span
                  style="display: inline-block; cursor: pointer;"
                >
                  <div
                    class="icon-container"
                    style="display: inline-block; width: 17px;"
                  >
                    <span
                      class="expanded-icon"
                    >
                      <svg
                        fill="currentColor"
                        style="vertical-align: top; color: rgb(150, 136, 62); height: 1em; width: 1em; padding-left: 2px;"
                        viewBox="0 0 15 15"
                      >
                        <path
                          d="M0 5l6 6 6-6z"
                        />
                      </svg>
                    </span>
                  </div>
                  <span
                    style="display: inline-block; color: rgb(210, 231, 228); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                  >
                    <span
                      class="object-key"
                    >
                      <span
                        style="vertical-align: top;"
                      >
                        "
                      </span>
                      <span>
                        root
                      </span>
                      <span
                        style="vertical-align: top;"
                      >
                        "
                      </span>
                    </span>
                    <span
                      style="display: inline-block; margin: 0px 5px; color: rgb(210, 231, 228); vertical-align: top;"
                    >
                      :
                    </span>
                  </span>
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(210, 231, 228);"
                  >
                    {
                  </span>
                </span>
                <div
                  class="object-meta-data"
                  style="display: inline-block; padding: 0px 0px 0px 10px;"
                >
                  <span
                    class="object-size"
                    style="color: rgb(95, 156, 146); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                  >
                    1 item
                  </span>
                  <span
                    class="copy-to-clipboard-container"
                  >
                    <span
                      style="cursor: pointer; display: inline;"
                    >
                      <span
                        class="copy-icon"
                      >
                        <svg
                          fill="currentColor"
                          preserveAspectRatio="xMidYMid meet"
                          style="vertical-align: top; color: rgb(62, 150, 91); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                          viewBox="0 0 40 40"
                        >
                          <g>
                            <path
                              d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </span>
              <div
                class="pushed-content object-container"
              >
                <div
                  class="object-content"
                  style="margin-left: 6px;"
                >
                  <div
                    class="object-key-val"
                    style="border-left: 1px solid #184E45; padding: 3px 5px 3px 20px;"
                  >
                    <span>
                      <span
                        style="display: inline-block; cursor: pointer;"
                      >
                        <div
                          class="icon-container"
                          style="display: inline-block; width: 17px;"
                        >
                          <span
                            class="collapsed-icon"
                          >
                            <svg
                              fill="currentColor"
                              style="vertical-align: top; color: rgb(76, 150, 62); height: 1em; width: 1em; padding-left: 2px;"
                              viewBox="0 0 15 15"
                            >
                              <path
                                d="M0 14l6-6-6-6z"
                              />
                            </svg>
                          </span>
                        </div>
                        <span
                          style="display: inline-block; color: rgb(210, 231, 228); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                        >
                          <span
                            class="object-key"
                          >
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                            <span>
                              response
                            </span>
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                          </span>
                          <span
                            style="display: inline-block; margin: 0px 5px; color: rgb(210, 231, 228); vertical-align: top;"
                          >
                            :
                          </span>
                        </span>
                        <span
                          style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(210, 231, 228);"
                        >
                          {
                        </span>
                      </span>
                    </span>
                    <span
                      class="brace-row"
                    >
                      <span
                        style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(210, 231, 228); padding-left: 0px;"
                      >
                        }
                      </span>
                      <div
                        class="object-meta-data"
                        style="display: inline-block; padding: 0px 0px 0px 10px;"
                      >
                        <span
                          class="object-size"
                          style="color: rgb(95, 156, 146); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                        >
                          0 items
                        </span>
                        <span
                          class="copy-to-clipboard-container"
                        >
                          <span
                            style="cursor: pointer; display: inline;"
                          >
                            <span
                              class="copy-icon"
                            >
                              <svg
                                fill="currentColor"
                                preserveAspectRatio="xMidYMid meet"
                                style="vertical-align: top; color: rgb(62, 150, 91); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                                viewBox="0 0 40 40"
                              >
                                <g>
                                  <path
                                    d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                  />
                                </g>
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="brace-row"
              >
                <span
                  style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(210, 231, 228); padding-left: 3px;"
                >
                  }
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
